src/datalinker/handlers/external.py

```diff
@@ -162,6 +163,9 @@ async def timeseries(
             pattern=ADQL_FOREIGN_COLUMN_REGEX,
         ),
     ] = None,
+    join_style: Annotated[
+        str, Query(title="Join style", pattern=TIMESERIES_JOIN_STYLE_REGEX)
+    ] = "ccdVisit",
```

Comment:
```suggestion
    join_style: Annotated[
        Literal["ccdVisit", "visit_detector"], Query(title="Join style")
    ] = "ccdVisit",
```
(and import `Literal` from `typing`). Then you can get rid of the regex stuff. (Or alternately you could use an enum; I'm agnostic on which is better.)